doc/design/OrionSTHSink.md

```diff
@@ -0,0 +1,208 @@
+#OrionSTHSink
+##Functionality
+`com.iot.telefonica.cygnus.sinks.OrionSTHSink`, or simply `OrionSTHSink` is a sink designed to persist NGSI-like context data events within a MongoDB server in an aggregated way. Usually, such a context data is notified by a [Orion Context Broker](https://github.com/telefonicaid/fiware-orion) instance, but could be any other system speaking the <i>NGSI language</i>.
+
+Independently of the data generator, NGSI context data is always [transformed](from_ngsi_events_to_flume_events.md) into internal Flume events at Cygnus sources thanks to `com.iot.telefonica.cygnus.handlers.OrionRestHandler`. In the end, the information within these Flume events must be mapped into specific MongoDB data structures.
+
+###Mapping Flume events to MongoDB data structures
+MongoDB organizes the data in databases that contain collections of Json documents. Such organization is exploited by `OrionSTHSink` each time a Flume event is taken, by performing the following workflow:
+
+1. The bytes within the event's body are parsed and a `NotifyContextRequest` object container is created.
+2. A database called as the `fiware-service` header value within the event is created (if not existing yet).
+3. The context responses/entities within the container are iterated, and a collection is created (if not yet existing) for each unit data. The unit data depends on the chosen data model (see the configuration section):
+    * <i>collection-per-service-path</i>: the collection is called as the `fiware-servicePath` header value within the event.
+    * <i>collection-per-entity</i>: the collection is called as the concatenation of the `fiware-servicePath`_`destination` headers values within the event.
+    * <i>collection-per-attribute</i>: the collection is called as the concatenation of the `fiware-servicePath`\_`destination`\_`attrName`.
+4. The context attributes within each context response/entity are iterated, and a new Json document is appended to the current collection.
+
+###Example
+Assuming the following Flume event is created from a notified NGSI context data (the code below is an <i>object representation</i>, not any real data format):
+
+    flume-event={
+        headers={
+	        content-type=application/json,
+	        fiware-service=vehicles,
+	        fiware-servicepath=4wheels,
+	        timestamp=1429535775,
+	        transactionId=1429535775-308-0000000000,
+	        ttl=10,
+	        destination=car1_car
+        },
+        body={
+	        entityId=car1,
+	        entityType=car,
+	        attributes=[
+	            {
+	                attrName=speed,
+	                attrType=kmh,
+	                attrValue=112.9
+	            },
+	            {
+	                attrName=oil_level,
+	                attrType=percentage,
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/fb6cabf7f0b9473ab675c0dfb93be6b828bcff57